2_process.R

```diff
@@ -43,15 +44,23 @@ p2_targets_list <- list(
      p2_sites_w_segs,
      get_site_flowlines(p1_reaches_sf, p2_site_list, sites_crs = 4269, max_matches = 1, search_radius = 0.1)),
   
-  # Pair PRMS segments with NHDPlusV2 reaches
+  # Pair PRMS segments with intersecting NHDPlusV2 reaches and contributing NHDPlusV2 catchments
   tar_target(
     p2_prms_nhdv2_xwalk,
     {
-      p1_reaches_sf %>%
+      # find NHDPlusV2 COMID's that intersect PRMS segments
+      reach_to_seg_xwalk <- p1_reaches_sf %>%
         split(.,.$subsegid) %>%
         purrr::map(.,pair_nhd_reaches,nhd_lines=p1_nhdv2reaches_sf) %>%
         purrr::map(.,summarize_paired_comids) %>%
         bind_rows()
+      
+      # find NHDPlusV2 COMID's that drain directly to PRMS segments
+      p1_reaches_sf %>%
+        split(.,.$subsegid) %>%
+        purrr::map(.,pair_nhd_catchments,prms_hrus=p1_catchments_sf,min_area_overlap=0.5,nhd_lines=p1_nhdv2reaches_sf,xwalk_table = reach_to_seg_xwalk) %>%
+        bind_rows() %>%
+        left_join(reach_to_seg_xwalk,.,by="PRMS_segid")
     }
```

Comment:
Might help to make this a separate function because it's long

Reply:
It may also be worth implementing targets parallelization for this. It took ~20 mins to complete on my laptop. I can file parallelization as an issue for the repo.

Reply:
I agree that this might be too long for the phase file. I was on the fence about creating a separate function and decided against it here so that you can see the functions that are used to create the target (`pair_nhd_reaches`,`summarize_paired_comids`, and `pair_nhd_catchments`). I could definitely make that change though.

Reply:
Sounds good! Don't worry about the parallelization. That's better for a different PR

Reply:
I didn't realize targets had options for parallelization but it'd be great to speed things up when possible (the crosswalk target takes ~15 min to build on my machine). Thanks for creating an issue - what do you think about implementing those changes in a separate PR that is just focused on parallelization/optimization changes?

Reply:
I opened up an issue for parallelization. One thing I/we need to check with the Data Science Branch is best practices for parallelizing targets that download from NWIS. I think Lindsay said they try not to do that, but it's just a vague memory for me. Do you know? I made a similar comment in the issue, so we can continue the discussion there